File: backend/src/main/java/com/tokiobank/transference/api/viewmodels/AccountViewModel.java
package com.tokiobank.transference.api.viewmodels;

import java.util.List;

import com.tokiobank.transference.domain.entities.AccountEntity;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class AccountViewModel{
	
	public AccountViewModel() {}
	public AccountViewModel(AccountEntity accountOrigin) {
		this.titular = accountOrigin.getTitular();
		this.numero = accountOrigin.getNumero();
	}
	@NotBlank(message = "Account identifilter is necessary")
	private String numero;
	private String titular;
	
	
	
	private List<AccountTransferenceViewModel> listAccountTransferenceOrigin;
	private List<AccountTransferenceViewModel> listAccountTransferenceDestiny;
}

File: frontend/src/app/core/viewmodel/account.viewmodel.ts
import { AccountTransference } from "./accountTransference.viewmodel";

    
export class Account {

    titular: string = '';
    numero: string = '';
    listAccountTransferenceOrigin!: AccountTransference[];
    listAccountTransferenceDestiny!: AccountTransference[];

    constructor(values: any = {}) {
        if (Object.entries(values).length === 0 && values.constructor === Object) {
          return;
        }

        if (values.listAccountTransferenceOrigin) {
            values.listAccountTransferenceOrigin = values.listAccountTransferenceOrigin.map((a: AccountTransference) => new AccountTransference(a));
        }

        if (values.listAccountTransferenceDestiny) {
          values.listAccountTransferenceDestiny = values.listAccountTransferenceDestiny.map((a: AccountTransference) => new AccountTransference(a));
        }

        Object.assign(this, values);
    }
}